### ISSUE TITLE
Is it possible to debug in VS Code using Dev Containers?

### ISSUE THREAD

**Issue by andwak** [bazelbuild/rules_dotnet#281]
Hi,

I have setup a project and the Bazel build works well. Bazel run also works.
I can create a debug build with the option '-c dbg' and attach to the executing process.

However, whenever I try to debug code I have written, VS code does not find the .cs file.
My assumption is, that I need to tell the compiler where the source code can be found.

I have had no success with 'sourceFileMap' in the launch.json file.
Maybe we need to specific the 'GenerateFullPaths' property for the compiler?

Did anybody manage to get this working?

Thanks for any pointers.

**Comment by andwak** [comment]
I carried out further investigation.

Bazel dotnet rules create a .param file containing the parameters for the csc.exe compilation call.
I no longer suspect this is related to /fullpaths (GenerateFullPaths) configurations (I added this property in a test branch).

When VS Code prompts to create the "file that is not found" and I manually replace this file with the source code file content, I can then debug. Obviously this is not a solution, just part of my analysis.

I then started to investigate the .pdb files that are created. I saw that the pdb contains a section "document" and this points to a folder where the assembly is executed. For example:

/root/.cache/bazel/_bazel_root/<xxx>/sandbox/processwrapper-sandbox/1237/execroot/__main__/<TargetFolder>/<MyCSharpFile>.cs

This appears to be handled by the csc.exe compiler.
I will try and analyse if this can be configured.
